content-src/components/ASRouterAdmin/ASRouterAdmin.jsx

```diff
@@ -229,23 +270,43 @@ export class DiscoveryStreamAdmin extends React.PureComponent {
   }
 
   render() {
+    const prefToggles = "enabled hardcoded_layout show_spocs personalized collapsible".split(
+      " "
+    );
     const { config, lastUpdated, layout } = this.props.state;
     return (
       <div>
-        <div className="dsEnabled">
-          <input
-            type="checkbox"
-            checked={config.enabled}
-            onChange={this.onEnableToggle}
-          />{" "}
-          enabled{" "}
-        </div>
+        <button className="button" onClick={this.restorePrefDefaults}>
+          Restore Pref Defaults
+        </button>{" "}
+        <button className="button" onClick={this.expireCache}>
+          Expire Cache
+        </button>
+        <table>
+          <tbody>
+            {prefToggles.map(pref => (
+              <Row key={pref}>
+                <td>
+                  <TogglePrefCheckbox
+                    checked={config[pref]}
+                    pref={pref}
+                    onChange={this.setConfigValue}
+                  />
+                </td>
+              </Row>
+            ))}
+          </tbody>
+        </table>
         <h3>Endpoint variant</h3>
         <p>
           You can also change this manually by changing this pref:{" "}
           <code>browser.newtabpage.activity-stream.discoverystream.config</code>
         </p>
-        <table style={config.enabled ? null : { opacity: 0.5 }}>
+        <table
+          style={
+            config.enabled && !config.hardcoded_layout ? null : { opacity: 0.5 }
```

Comment:
in false condition we are just reducing opacity but the radio button stays clickable and clears cache though layout stays same.

Reply:
Also noticed, we are replacing $apikey with key value in layout_endpoint when switching between layouts, not sure if it matters as its dev tool

Reply:
I think I'm ok with "in false condition we are just reducing opacity but the radio button stays clickable and clears cache though layout stays same." it's also just reusing the existing way we disabled endpoint if enable/disable is toggled. I think as long as it's clear that changing the layout if it's grey has no effect, that's probably fine. Also technically you CAN change your endpoint if layout is disabled in the pref.

I'm not sure what you mean by "we are replacing $apikey with key value in layout_endpoint when switching between layouts, not sure if it matters as its dev tool" When I switch between layouts the pref is keeping the proper value in the url?

Reply:
Oh I see what you mean by "we are replacing $apikey with key value in layout_endpoint when switching between layouts, not sure if it matters as its dev tool"

Hm, I think that's probably fine? It's also not something I'm adding so probably if it's not fine, it's another issue.